anta/tests/vxlan.py

```diff
@@ -73,3 +82,95 @@ def test(self) -> None:
+    This test verifies the VNI-VLAN bindings of the Vxlan1 interface.
+
+    Expected Results:
+      * success: The test will pass if the VNI-VLAN bindings provided are properly configured.
+      * failure: The test will fail if any VNI lacks bindings or if any bindings are incorrect.
+      * skipped: The test will be skipped if the Vxlan1 interface is not configured.
+    """
+
+    name = "VerifyVxlanVniBinding"
+    description = "Verifies the VNI-VLAN bindings of the Vxlan1 interface"
+    categories = ["vxlan"]
+    commands = [AntaCommand(command="show vxlan vni", ofmt="json")]
+
+    class Input(AntaTest.Input):  # pylint: disable=missing-class-docstring
+        bindings: Dict[Vni, Vlan]
+        """VNI to VLAN bindings to verify"""
+
+    @AntaTest.anta_test
+    def test(self) -> None:
+        self.result.is_success()
+
+        no_binding = []
+        wrong_binding = []
+
+        if (vxlan1 := get_value(self.instance_commands[0].json_output, "vxlanIntfs.Vxlan1")) is None:
+            self.result.is_skipped("Vxlan1 interface is not configured")
+            return
+
+        for vni, vlan in self.inputs.bindings.items():
+            vni = str(vni)
+            if vni in vxlan1["vniBindings"]:
+                vlan_ = vxlan1["vniBindings"][vni]["vlan"]
+            elif vni in vxlan1["vniBindingsToVrf"]:
+                vlan_ = vxlan1["vniBindingsToVrf"][vni]["vlan"]
+            else:
+                no_binding.append(vni)
+                vlan_ = None
+
+            if vlan_ and vlan != vlan_:
+                wrong_binding.append({vni: vlan_})
+
+        if no_binding:
+            self.result.is_failure(f"The following VNI(s) have no binding: {no_binding}")
+
+        if wrong_binding:
+            self.result.is_failure(f"The following VNI(s) have the wrong VLAN binding: {wrong_binding}")
+
+
+class VerifyVxlanVtep(AntaTest):
+    """
+    This test verifies the VTEP peers of the Vxlan1 interface.
+
+    Expected Results:
+      * success: The test will pass if all provided VTEP peers are identified.
+      * failure: The test will fail if any VTEP peer is missing.
+      * skipped: The test will be skipped if the Vxlan1 interface is not configured.
+    """
+
+    name = "VerifyVxlanVtep"
+    description = "Verifies the VTEP peers of the Vxlan1 interface"
+    categories = ["vxlan"]
+    commands = [AntaCommand(command="show vxlan vtep", ofmt="json")]
+
+    class Input(AntaTest.Input):  # pylint: disable=missing-class-docstring
+        vteps: List[IPv4Address]
+        """List of VTEP peers to verify"""
+
+    @AntaTest.anta_test
+    def test(self) -> None:
+        missing_vteps = []
+
+        if (vxlan1 := get_value(self.instance_commands[0].json_output, "interfaces.Vxlan1")) is None:
+            self.result.is_skipped("Vxlan1 interface is not configured")
+            return
+
+        if len(vteps := vxlan1["vteps"]) == 0:
+            self.result.is_failure("Vxlan1 interface has no VTEP")
+            return
```

Comment:
what if `input.vteps` is an empty list ? Maybe someone wants to verify there is no VTEP peers :)

Reply:
Fixed in 60aa0117abf83f27faed801b5710aaffb0cab147